resilience4j-core/src/main/java/io/github/resilience4j/core/functions/Either.java

```diff
@@ -0,0 +1,239 @@
+/*
+ *
+ *  Copyright 2021 krnSaurabh
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package io.github.resilience4j.core.functions;
+
+import java.io.Serializable;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.Function;
+
+/*
+ * Either represents a value of two possible types. An Either is either a {@link Left} or a
+ * {@link Right}.
+ * This interface is similar to {@link io.vavr.control.Either}.
+ */
+public interface Either<L, R> {
```

Comment:
@RobWin I have created our custom `Either` that replaces `vavr Either`. This class has some utility methods taken from `vavr Either` . Let me know if this approach is fine.